wireapp/wire-ios-mono
# chore: split unit test plan

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

The project has a `Wire-iOS Test` folder with a mixture of unit and snapshot tests. As developer this makes it difficult and time consuming just to write and run a few unit tests. It also has downsides, that you can execute the tests only on certain simulator devices.

### Solutions

Thus I've created a new `Wire-iOS UnitTest` target, included it in the existing test plans and added a new `UnitTests.xctestplan` to run only unit tests within a few seconds.

To make it work I've created a new folder "Tests", that contains shared code for mocking and fixtures like the massive `CoreDataFixture`. Also the `xctestplan` files moved there.

<img width="308" alt="Screenshot 2023-11-14 at 08 01 20" src="https://github.com/wireapp/wire-ios/assets/7196958/9624a17f-75c4-45ae-80ac-aa4551122ed0">


### Known issue

- Not all unit tests have been moved to the new target yet. Why? Because many unit tests have shared resources with snapshot tests and extensions on Apple classes etc. and it is hard to split without duplicating everything and time consuming. To make a first step I've only migrated a few tests.

### Testing

#### How to Test

- In Xcode select your testplan and run it with CMD+U 🥳
- On CI nothing should have changed, because the `xctestplan` files have been updated and contain the same tests as before.

### Notes

- `testThatSharedInstanceReturnQualityTypeWifi` seems to be dependent on your internet connection and doesn't work when you are offline.

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
